SSA/Experimental/Bits/Fast/Tactic.lean

```diff
@@ -141,6 +120,27 @@ partial def first_rep (w : Q(Nat)) (e : Q( BitStream)) : SimpM (Σ (x : Q(BitStr
         q($anext + $bnext),
         .app (.app (.app (.app (.app (.app (.app (.const ``BitStream.add_congr []) w) a) anext) b) bnext) aproof) bproof
       ⟩
+    | ~q(@HAnd.hAnd BitStream BitStream BitStream _ $a $b) => do
+      let ⟨ anext, aproof ⟩ ← first_rep w a
+      let ⟨ bnext, bproof ⟩ ← first_rep w b
+      return ⟨
+        q($anext &&& $bnext),
+        .app (.app (.app (.app (.app (.app (.app (.const ``BitStream.and_congr []) w) a) anext) b) bnext) aproof) bproof
```

Comment:
Not blocking this, but surely this can be simplified with the `mkAppM` and friends?